test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -36,6 +51,22 @@
        (let [catalog (testutils/get-catalog)]
          (is (testutils/catalog-contains? catalog "Notify" "this command echoes a thing\n")))))))
 
+(deftest ^:integration test-complicated-external-command-execution
+  (testing "puppet functions can call more complicated external commands successfully"
+    ; The generate puppet function runs a fully qualified command with arguments.
+    ; This function calls into Puppet::Util::Execution.execute(), which calls into
+    ; our shell-utils code via Puppet::Util::ExecutionStub which we call in
+    ; Puppet::Server::Execution.
+    (testutils/write-site-pp-file
+     (format "$a = generate('%s', '-c', \"echo foo\"); notify {$a: }"
+             "/bin/sh"))
```

Comment:
Chatted to @haus about whether we needed to have a wrapper script around this - we decided that it was most likely fine, and if it were to fail anywhere it would probably be in travis since it uses Docker.